main.nf

```diff
@@ -177,75 +160,380 @@ process get_software_versions {
             if (filename.indexOf(".csv") > 0) filename
             else null
         }
+    label 'process_low'
+
 
     output:
     file 'software_versions_mqc.yaml' into software_versions_yaml
     file "software_versions.csv"
+    file "*.txt"
 
     script:
-    // TODO nf-core: Get all tools to print their version number here
     """
     echo $workflow.manifest.version > v_pipeline.txt
     echo $workflow.nextflow.version > v_nextflow.txt
-    fastqc --version > v_fastqc.txt
+    fastqc --version &> v_fastqc.txt
+    samtools --version > v_samtools.txt
     multiqc --version > v_multiqc.txt
     scrape_software_versions.py &> software_versions_mqc.yaml
     """
 }
 
 /*
- * STEP 1 - FastQC
+ * STEP 0: Extract reads mapping to specific chromosome(s)
  */
-process fastqc {
-    tag "$name"
+if (params.chr){
+  process extractReadsMappingToChromosome{
+    tag "${name}.${chr_list_joined}"
     label 'process_medium'
-    publishDir "${params.outdir}/fastqc", mode: 'copy',
-        saveAs: { filename -> filename.indexOf(".zip") > 0 ? "zips/$filename" : "$filename" }
-
+  
     input:
-    set val(name), file(reads) from read_files_fastqc
+    set val(name), file(bam) from bam_chr 
 
     output:
-    file "*_fastqc.{zip,html}" into fastqc_results
+    set val("${name}.${chr_list_joined}"), file("${name}.${chr_list_joined}.bam") into bam_files_check
+                
 
     script:
+    //If multiple chr were specified, then join space separated list for naming: chr1 chr2 -> chr1_chr2, also resolve region specification with format chr:start-end
+    chr_list_joined = params.chr.split(' |-|:').size() > 1 ? params.chr.split(' |-|:').join('_') : params.chr
     """
-    fastqc --quiet --threads $task.cpus $reads
+    samtools index -@$task.cpus $bam 
+    samtools view -hb $bam ${params.chr} -@$task.cpus -o ${name}.${chr_list_joined}.bam
     """
+  }
 }
 
 /*
- * STEP 2 - MultiQC
+ * STEP 1: Check for paired-end or single-end bam
  */
-process multiqc {
-    publishDir "${params.outdir}/MultiQC", mode: 'copy'
+process checkIfPairedEnd{
+  tag "$name"
+  label 'process_low'
+  input:
+  set val(name), file(bam) from bam_files_check
+
+  output:
+  set val(name), file(bam), file('*paired.txt') optional true into bam_files_paired_map_map,
+                                                                   bam_files_paired_unmap_unmap,
+                                                                   bam_files_paired_unmap_map,
+                                                                   bam_files_paired_map_unmap
+  set val(name), file(bam), file('*single.txt') optional true into bam_file_single_end // = is not paired end
+ 
+
+  script:
+  """
+  # Take samtools header + the first 1000 reads (to safe time, otherwise also all can be used) and check whether for 
+  # all, the flag for paired-end is set. Compare: https://www.biostars.org/p/178730/ . 
+
```

Comment:
```suggestion
  // Take samtools header + the first 1000 reads (to safe time, otherwise also all can be used) and check whether for 
  // all, the flag for paired-end is set. Compare: https://www.biostars.org/p/178730/ . 
  """
```

Reply:
Makes it a bit easier to read.